src/csm_testing/tests/sat_functional/firmware/test_firmware.py

```diff
@@ -0,0 +1,188 @@
+
+        # Now delete the snapshot to keep the system clean
+        delete_command = f"sat firmware --delete-snapshot {snapshot_name}"
+        proc = subprocess.run(shlex.split(delete_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        delete_output = proc.stderr.decode().strip()
+        expected_msg = f"INFO: Snapshot {snapshot_name} deleted successfully."
+        self.assertIn(expected_msg, delete_output)
+
+    def test_firmware_describe_existing_snapshot(self):
+        """Test that `sat firmware --snapshots SNAPSHOT` describes an existing firmware snapshot."""
+        # Get the list of available snapshots
+        list_command = "sat firmware --snapshots"
+        proc = subprocess.run(shlex.split(list_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        snapshots = [line.strip() for line in proc.stdout.decode().splitlines() if line.strip() and not line.startswith("#")]
+        self.assertGreater(len(snapshots), 0, "No firmware snapshots available for testing.")
+
+        # Query the first snapshot
+        snapshot = snapshots[0]
+        query_command = f"sat firmware --snapshots {snapshot}"
+        proc = subprocess.run(shlex.split(query_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        output = proc.stdout.decode().strip()
+
+        # Check that the output contains the snapshot name and the expected table header
+        self.assertIn(snapshot, output)
+        for col in SAT_FIRMWARE_HEADER:
+            self.assertIn(col, output)
+
+    def test_firmware_query_multiple_snapshots(self):
+        """Test that `sat firmware --snapshots SNAPSHOT_1 SNAPSHOT_2` prints the details for two snapshots."""
+        # Get the list of available snapshots
+        list_command = "sat firmware --snapshots"
+        proc = subprocess.run(shlex.split(list_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        snapshots = [line.strip() for line in proc.stdout.decode().splitlines() if line.strip() and not line.startswith("#")]
+        self.assertGreaterEqual(len(snapshots), 2, "Less than two firmware snapshots available for testing.")
+
+        # Query the first two snapshots
+        snap1, snap2 = snapshots[0], snapshots[1]
+        query_command = f"sat firmware --snapshots {snap1} {snap2}"
+        proc = subprocess.run(shlex.split(query_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        output = proc.stdout.decode()
+
+        # Check that both snapshot names and table headers are present in the output
+        self.assertIn(snap1, output)
+        self.assertIn(snap2, output)
+        for col in SAT_FIRMWARE_HEADER:
+            self.assertIn(col, output)
+
+    def test_firmware_query_nonexistent_snapshot(self):
+        """Test that `sat firmware --snapshots NON_EXISTENT_SNAPSHOT` querying a non-existent firmware snapshot."""
+        non_existent_snapshot = "NON_EXISTENT_SNAPSHOT"
```

Comment:
Probably safe, but it might be better to generate a random and basically guaranteed to be unique value here (e.g. using the `uuid` module)

```
>>> import uuid
>>> str(uuid.uuid4())
'2713b54e-3687-44b1-8387-8f83fd85a5ab'
```

That's assuming that such a value is a valid FAS snapshot name, which I have not checked.